openml/datasets/functions.py

```diff
@@ -621,28 +659,74 @@ def _create_dataset_from_description(description, features, qualities, arff_file
-        description.get("oml:licence"),
-        description["oml:url"],
-        description.get("oml:default_target_attribute"),
-        description.get("oml:row_id_attribute"),
-        description.get("oml:ignore_attribute"),
-        description.get("oml:version_label"),
-        description.get("oml:citation"),
-        description.get("oml:tag"),
-        description.get("oml:visibility"),
-        description.get("oml:original_data_url"),
-        description.get("oml:paper_url"),
-        description.get("oml:update_comment"),
-        description.get("oml:md5_checksum"),
+        data_format=description["oml:format"],
+        dataset_id=description["oml:id"],
+        version=description["oml:version"],
+        creator=description.get("oml:creator"),
+        contributor=description.get("oml:contributor"),
+        collection_date=description.get("oml:collection_date"),
+        upload_date=description.get("oml:upload_date"),
+        language=description.get("oml:language"),
+        licence=description.get("oml:licence"),
+        url=description["oml:url"],
+        default_target_attribute=description.get(
+            "oml:default_target_attribute"
+        ),
+        row_id_attribute=description.get("oml:row_id_attribute"),
+        ignore_attribute=description.get("oml:ignore_attribute"),
+        version_label=description.get("oml:version_label"),
+        citation=description.get("oml:citation"),
+        tag=description.get("oml:tag"),
+        visibility=description.get("oml:visibility"),
+        original_data_url=description.get("oml:original_data_url"),
+        paper_url=description.get("oml:paper_url"),
+        update_comment=description.get("oml:update_comment"),
+        md5_checksum=description.get("oml:md5_checksum"),
         data_file=arff_file,
         features=features,
-        qualities=qualities)
+        qualities=qualities,
+    )
     return dataset
+
+
+def _get_online_dataset_arff(did):
+    """Download the arff file for a given dataset id
+    from the OpenML website.
+
+    Parameters
+    ----------
+    did : int
+        A dataset id.
+
+    Returns
+    -------
+    str
+        A string representation of an arff file.
+    """
+    dataset_xml = openml._api_calls._perform_api_call("data/%d" % did)
+    # build a dict from the xml.
+    # use the url from the dataset description and return the arff string
+    return openml._api_calls._read_url(
+        xmltodict.parse(dataset_xml)['oml:data_set_description']['oml:url']
+    )
+
+
+def _get_online_dataset_format(did):
+    """Get the dataset format for a given dataset id
+    from the OpenML website.
+
+    Parameters
+    ----------
+    did : int
+        A dataset id.
+
+    Returns
+    -------
+    str
+        Dataset format.
+    """
+    dataset_xml = openml._api_calls._perform_api_call("data/%d" % did)
+    # build a dict from the xml and get the format from the dataset description
+    return xmltodict\
```

Comment:
I am not a big fan of `\`. I would instead break the line in the paranthesis or reaffect the name of the keys into 2 variables. But this is more personnal thing I assume

Reply:
Neither am I. Just trying to keep it as readable as possible.